services/net/filemonitor/FilemonitorAction.cs

```diff
@@ -0,0 +1,678 @@
+        return await this.Api.UpdateContentReferenceAsync(reference);
+    }
+
+    /// <summary>
+    /// Get a news item value (e.g. headline, published date, author) from an XML story. Different publications store these values
+    /// in different tags/attributes in the XML document. The keys for the list of supported values is in the Fields class.
+    ///
+    /// Each of these keys maps to an entry in the data source's Connection string which contains the tag/attribute name. If the
+    /// value is stored in a tag, the Connection string entry contains the name of the tag, e.g., for the GLOBE data source
+    /// the story content is in the body.content tag. For the BCNG data source it is in the story tag, so the connection
+    /// entries for those two items are:
+    ///
+    ///     "story": "body.content"
+    ///     "story": "story"
+    ///
+    /// respectively. If the value is stored in a tag's attribute, the Connection string value will have the following format:
+    ///
+    ///     tag-name!attribute-name
+    ///
+    /// For the GLOBE data source the publication date is stored in the date.publication attribute of the pubdata tag for each
+    /// story. This results in the following entry for "date" in the connection string for GLOBE:
+    ///
+    ///     "date": "pubdata!date.publication"
+    ///
+    /// </summary>
+    /// <param name="story"></param>
+    /// <param name="key"></param>
+    /// <param name="dataSource"></param>
+    /// <returns></returns>
+    private string GetXmlData(XmlElement story, string key, DataSourceModel dataSource)
+    {
+        var value = GetConnectionValue(dataSource, key);
+        var result = "";
+
+        if (!String.IsNullOrEmpty(value))
+        {
+            var comps = value.Split('!');
+            if (comps.Count() == 2) // Tag and attribute name present
+            {
+                XmlNodeList nodeList = story.GetElementsByTagName(comps[0]);
+
+                if (nodeList is not null && nodeList.Count > 0 && nodeList[0]!.Attributes is not null && nodeList![0]!.Attributes!.Count > 0)
+                {
+                    result = nodeList![0]!.Attributes![comps[1]!]!.Value;
+                }
+                else
+                {
+                    throw new InvalidOperationException($"No node/attribute pair in story XML matching '{comps[0]}/{comps[1]}' for data source '{dataSource.Code}.");
+                }
+            }
+            else
+            if (comps.Count() == 1) // Tag name only
+            {
+                try
+                {
+                    XmlNodeList nodeList = story.GetElementsByTagName(value);
+                    result = nodeList is not null && nodeList.Count > 0 ? nodeList[0]!.InnerText : "";
+                }
+                catch (Exception e)
+                {
+                    _logger.LogInformation(e, $"Error extracting node '{key}' for data source {dataSource.Code}.");
```

Comment:
I only wanted to log the error at this point as it would be an issue with the input file. In the previous statement, where the exception is thrown, it's more likely to be a data source configuration error, so it will happen for every story.

Reply:
You're hiding errors from the caller and you're not logging them as errors.  Both are problematic.  Sometimes an exception shouldn't bubble up to the caller because it is handled by the function.  Is this error something the caller needs to know about?  Or is it being ignored.  If by change it can be ignored, then `LogWarning`.

Reply:
Ahhhh...ok. I figured the caller didn't need to know about this. I certainly didn't want this situation to add to the data source's failed_attempts counter. I can change it from LogInformation to LogWarning.